lib/spack/spack/package.py

```diff
@@ -936,6 +957,10 @@ def build_process():
             self._total_time = time.time() - start_time
             build_time = self._total_time - self._fetch_time
 
+            # Symlink local license to global license if one is required
+            if self.license_required and self.license_files:
+                self.symlink_license()
```

Comment:
Can this be implemented as a post-install hook?  It would keep the logic a bit more separate.  If it were implemented as a post-install hook, it would also ensure that the build is cleaned up properly if anything goes wrong here in symlink_license.